lib/Algo2023Day04.cs

```diff
@@ -8,32 +10,112 @@ namespace Lib
-                string[] playNumbers = line.Split(':')[1].Split('|')[1].Split(' ');
-                playNumbers = playNumbers.Where(e => e != "").ToArray();
-                
-                foreach (string playNumber in playNumbers)
+                return bonus(input);
+            }
+            else
+            {
+                return standard(input);
+            }
+
+            string standard(string[] input)
+            {
+                int totalPoint = 0;
+                foreach (var line in input) 
                 {
-                    if (winningNumbers.Contains(playNumber))
+                    int gamePoint = 0;
+                    string[] winningNumbers = line.Split(':')[1].Split('|')[0].Split(' ');
+                    winningNumbers = winningNumbers.Where(e => e != "").ToArray();
+                    string[] playNumbers = line.Split(':')[1].Split('|')[1].Split(' ');
+                    playNumbers = playNumbers.Where(e => e != "").ToArray();
+                
+                    foreach (string playNumber in playNumbers)
                     {
-                        gamePoint *= 2;
-                        if(gamePoint == 0)
+                        if (winningNumbers.Contains(playNumber))
                         {
-                            gamePoint++;
+                            gamePoint *= 2;
+                            if(gamePoint == 0)
+                            {
+                                gamePoint++;
+                            }
                         }
                     }
+                    totalPoint += gamePoint;
+
                 }
-                totalPoint += gamePoint;
+                return totalPoint.ToString();
+            }
+            string bonus(string[] input)
+            {
+
+                List<Card> cards = new List<Card>();
+                foreach (var line in input)
+                {
+                    string[] cardGame = line.Split(':')[0].Split(' ');
+                    int cardId = int.Parse(cardGame.Where(e => e != "").ToArray()[1]);
+                    List<int> winningNumbers = line
+                        .Split(':')[1]
+                        .Split('|')[0]
+                        .Split(' ')
+                        .ToList()
+                        .Where(str => !string.IsNullOrWhiteSpace(str))
+                        .Select( str => int.Parse(str))
+                        .ToList();
+                    List<int> playNumbers = line
+                        .Split(':')[1]
+                        .Split('|')[1]
+                        .Split(' ')
+                        .ToList()
+                        .Where(str => !string.IsNullOrWhiteSpace(str))
+                        .Select(str => int.Parse(str))
+                        .ToList();
+                    Card card = new Card(cardId, winningNumbers, playNumbers);
+                    cards.Add(card);
+                }
+                for(int i = 1;i <= 202;i ++)
```

Comment:
magic values
recherche les référence min et max pour les bornes du for

```suggestion
                for(int i = 1;i <= 202;i ++)
```